lib/model/model.go

```diff
@@ -1465,12 +1465,10 @@ func sendIndexTo(minSequence int64, conn protocol.Connection, folder string, fs
 func (m *Model) updateLocalsFromScanning(folder string, fs []protocol.FileInfo) {
 	m.updateLocals(folder, fs)
 
-	// Fire the LocalChangeDetected event to notify listeners about local
-	// updates.
+	// Fire the LocalChangeDetected event to notify listeners about local updates.
 	m.fmut.RLock()
-	path := m.folderCfgs[folder].Path()
+	m.localChangeDetected(m.folderCfgs[folder], fs)
 	m.fmut.RUnlock()
-	m.localChangeDetected(folder, path, fs)
```

Comment:
I'd rather it got what it needed under the lock, and passed into the function outside of the lock. If there are a lot of files, the lock could be pointlessly held for long periods of time.